src/EditorFeatures/Text/Extensions.cs

```diff
@@ -73,7 +88,19 @@ public static IEnumerable<Document> GetRelatedDocumentsWithChanges(this ITextSna
         /// associated with the buffer if it is linked into multiple projects or is part of a Shared Project. In this case, the <see cref="Workspace"/>
         /// is responsible for keeping track of which of these <see cref="Document"/>s is in the current project context.
         /// </summary>
+        [Obsolete("Use GetDocument instead")]
         public static Document GetOpenDocumentInCurrentContextWithChanges(this ITextSnapshot text)
+            => text.GetDocument();
+
+        /// <summary>
+        /// Gets the <see cref="Document"/> from the corresponding <see cref="Workspace.CurrentSolution"/> that is associated with the <see cref="ITextSnapshot"/>'s buffer
+        /// in its current project context, updated to contain the same text as the snapshot if necessary. There may be multiple <see cref="Document"/>s
+        /// associated with the buffer if it is linked into multiple projects or is part of a Shared Project. In this case, the <see cref="Workspace"/>
+        /// is responsible for keeping track of which of these <see cref="Document"/>s is in the current project context.
+        /// 
+        /// Use <see cref="GetDocuments"/> to retrieve all potential <see cref="Document"/>s for a given <see cref="ITextSnapshot"/>.
+        /// </summary>
+        public static Document GetDocument(this ITextSnapshot text)
             => text.AsText().GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
Not renaming the SourceText extension as GetDocument either?

Reply:
I cared most about the public surface area.  i can change the internal APIs as well if you'd like :)

Reply:
Let's keep things consistent.

Reply:
sounds good.